src/scripts/main.js

```diff
@@ -1 +1,72 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const handleClick = () => {
+    resolve('First promise was resolved');
+    clearTimeout(timeout);
+    document.removeEventListener('click', handleClick);
+  };
+
+  document.addEventListener('click', handleClick);
+
+  const timeout = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+    document.removeEventListener('click', handleClick);
+  }, 3000);
+});
+
+const secondPromise = new Promise((resolve) => {
+  const handleClick = () => {
+    resolve('Second promise was resolved');
+  };
+
+  document.addEventListener('click', handleClick);
+  document.addEventListener('contextmenu', handleClick);
```

Comment:
Consider removing the event listeners after the promise is resolved to prevent potential memory leaks or unexpected behavior. This can be done by calling `document.removeEventListener('click', handleClick);` and `document.removeEventListener('contextmenu', handleClick);` after resolving the promise.